src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -344,6 +344,44 @@ impl<'a, T: Timestamp + Lattice> StorageControllerMut<'a, T> {
                 );
         }
     }
+    async fn recv(&mut self) -> Result<Option<StorageResponse<Self::Timestamp>>, anyhow::Error> {
+        self.state.client.recv().await
+    }
+}
+
+// Internal interface
+impl<T: Timestamp + Lattice> Controller<T> {
+    /// Create a new storage controller from a client it should wrap.
+    pub fn new(client: Box<dyn StorageClient<T>>) -> Self {
+        Self {
+            state: StorageControllerState::new(client),
+        }
+    }
+
+    /// Validate that a collection exists for all identifiers, and error if any do not.
+    pub(super) fn validate_ids(
```

Comment:
why do we need `pub(super)` here? I think for this and `allow_compaction` only need to be private, as they appear to only be used in the trait impl

Reply:
We don't, I'll remove it